File: go/src/iterative_sorts/insertion_sort.go
package iterativesorts

func InsertionSort(arr []int) {
  for i := 1; i < len(arr); i++ {
    numberToInsert := arr[i]
    j := i

    for ; j > 0; j-- {
      if numberToInsert >= arr[j-1] {
        break
      }
      arr[j] = arr[j-1]
    }

    arr[j] = numberToInsert
  }
}

File: rust/src/iterative_sorts/insertion_sort.rs
pub fn insertion_sort(mut nums: Vec<i32>) -> Vec<i32> {
    for i in 1..nums.len() {
        let number_to_insert = nums[i];
        let mut j = i;

        while j > 0 {
            if number_to_insert >= nums[j - 1] {
                break;
            }
            nums[j] = nums[j - 1];
            j -= 1;
        }

        nums[j] = number_to_insert;
    }

    return nums;
}
